src/resource_clients/dataset.ts

```diff
@@ -163,6 +163,55 @@ export class DatasetClient<
         return undefined;
     }
 
+    /**
+     * Returns a URL that can be used to access dataset items.
+     *
+     * If the token has permission to access the dataset's URL signing key,
+     * the URL will include a signature to verify its authenticity.
+     *
+     * You can optionally control how long the signed URL should be valid using the `expiresInMillis` option.
+     * This value sets the expiration duration in milliseconds from the time the URL is generated.
+     * If not provided, the URL will not expire.
+     *
+     * Any other options (like `limit` or `prefix`) will be included as query parameters in the URL.
+     *
+     */
+    async getPublicItemsUrl(options: DatasetClientListItemOptions = {}, expiresInMillis?: number): Promise<string> {
```

Comment:
```suggestion
    async createPublicItemsUrl(options: DatasetClientListItemOptions = {}, expiresInMillis?: number): Promise<string> {
```

Maybe call it _create_ to make it explicit that the URL is actually being crafted on the go? And also to distinguish it from the already provided `publicItemsUrl`?

I'd still wouldn't be surprised if users were a bit confused by this 😄